internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -74,3 +80,129 @@ func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationR
 		}
 	})
 }
+
+// handleInstrumentationCR manages OpenTelemetry Instrumentation CRs based on DSCI monitoring configuration
+func handleInstrumentationCR(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	dsci, err := cluster.GetDSCI(ctx, rr.Client)
+	if err != nil {
+		return fmt.Errorf("failed to get DSCInitialization instance: %w", err)
+	}
+
+	// Get the monitoring instance
+	monitoring, ok := rr.Instance.(*serviceApi.Monitoring)
+	if !ok {
+		return fmt.Errorf("instance is not of type *serviceApi.Monitoring")
+	}
+
+	instrumentationName := "opendatahub-instrumentation"
+	instrumentationNamespace := monitoring.Spec.Namespace
+
+	switch dsci.Spec.Monitoring.ManagementState {
+	case operatorv1.Managed:
+		// Create or update the instrumentation CR
+		return createOrUpdateInstrumentationCR(ctx, rr.Client, instrumentationName, instrumentationNamespace, dsci)
+	case operatorv1.Unmanaged, operatorv1.Removed:
+		// Remove the instrumentation CR if it exists
+		return removeInstrumentationCR(ctx, rr.Client, instrumentationName, instrumentationNamespace)
+	default:
+		return fmt.Errorf("unsupported monitoring management state: %s", dsci.Spec.Monitoring.ManagementState)
+	}
+}
+
+// createOrUpdateInstrumentationCR creates or updates the OpenTelemetry Instrumentation CR using unstructured resources
+func createOrUpdateInstrumentationCR(ctx context.Context, cli client.Client, name, namespace string, dsci *dsciv1.DSCInitialization) error {
+	// Set default sample ratio if not specified
+	sampleRatio := "0.1"
+	if dsci.Spec.Monitoring.Traces != nil && dsci.Spec.Monitoring.Traces.SampleRatio != nil {
+		sampleRatio = *dsci.Spec.Monitoring.Traces.SampleRatio
+	}
+
+	// Define the GVK for Instrumentation
+	gvk := schema.GroupVersionKind{
```

Comment:
maybe add this to the [gvk package](https://github.com/opendatahub-io/opendatahub-operator/blob/main/pkg/cluster/gvk/gvk.go)